### ISSUE TITLE
"from" keyword is not escaped properly during normalization

### ISSUE THREAD

**Issue by jrhizor** [airbytehq/airbyte#1773]
## Expected Behavior
A sync for the hubspot source should be able to normalize the `email_events` stream, which includes this field:
```
            "from": {
              "type": ["null", "string"]
            },
```

## Current Behavior
This happens on at least Snowflake and Postgres:
```
2021-01-21 22:45:16 DEBUG (/tmp/workspace/26/0) DockerProcessBuilderFactory(create):104 - Preparing command: docker run --rm -i -v airbyte_workspace:/data -v /tmp/airbyte_local:/local -w /data/26/0/normalize --network host airbyte/normalization:0.1.8 run --integration-type postgres --config target_config.json --catalog catalog.json
2021-01-21 22:45:16 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - Namespace(config='target_config.json', integration_type=<DestinationType.postgres: 'postgres'>, out='/data/26/0/normalize')
2021-01-21 22:45:16 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - transform_postgres
2021-01-21 22:45:16 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - Processing catalog.json...
2021-01-21 22:45:16 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - Generating email_events.sql in /data/26/0/normalize/models/generated/
2021-01-21 22:45:16 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - Generating forms.sql in /data/26/0/normalize/models/generated/
2021-01-21 22:45:18 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - Running with dbt=0.18.1
2021-01-21 22:45:19 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - Found 2 models, 0 tests, 0 snapshots, 0 analyses, 308 macros, 0 operations, 0 seed files, 2 sources
2021-01-21 22:45:19 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 -
2021-01-21 22:45:19 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - 22:45:19 | Concurrency: 32 threads (target='prod')
2021-01-21 22:45:19 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - 22:45:19 |
2021-01-21 22:45:19 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - 22:45:19 | 1 of 2 START table model public.email_events................................................................. [RUN]
2021-01-21 22:45:19 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - 22:45:19 | 2 of 2 START table model public.forms........................................................................ [RUN]
2021-01-21 22:45:20 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - 22:45:20 | 1 of 2 ERROR creating table model public.email_events........................................................ [[31mERROR[0m in 0.31s]
2021-01-21 22:45:20 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - 22:45:20 | 2 of 2 OK created table model public.forms................................................................... [[32mSELECT 11[0m in 0.32s]
2021-01-21 22:45:20 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - 22:45:20 |
2021-01-21 22:45:20 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - 22:45:20 | Finished running 2 table models in 0.65s.
2021-01-21 22:45:20 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 -
2021-01-21 22:45:20 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - [31mCompleted with 1 error and 0 warnings:[0m
2021-01-21 22:45:20 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 -
2021-01-21 22:45:20 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - [33mDatabase Error in model email_events (models/generated/email_events.sql)[0m
2021-01-21 22:45:20 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - syntax error at or near "from"
2021-01-21 22:45:20 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - LINE 87: ), '') || '-' || coalesce(cast(from as
2021-01-21 22:45:20 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - ^
2021-01-21 22:45:20 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - compiled SQL at ../build/run/airbyte_utils/models/generated/email_events.sql
2021-01-21 22:45:20 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 -
2021-01-21 22:45:20 INFO (/tmp/workspace/26/0) LineGobbler(voidCall):69 - Done. PASS=1 WARN=0 ERROR=1 SKIP=0 TOTAL=2
2021-01-21 22:45:20 DEBUG (/tmp/workspace/26/0) DefaultNormalizationRunner(close):97 - Closing tap process
2021-01-21 22:45:20 ERROR (/tmp/workspace/26/0) DefaultSyncWorker(run):117 - Normalization Failed.
io.airbyte.workers.WorkerException: Normalization Failed.
at io.airbyte.workers.DefaultSyncWorker.run(DefaultSyncWorker.java:114) [io.airbyte-airbyte-workers-0.12.1-alpha.jar:?]
at io.airbyte.workers.DefaultSyncWorker.run(DefaultSyncWorker.java:48) [io.airbyte-airbyte-workers-0.12.1-alpha.jar:?]
at io.airbyte.workers.wrappers.OutputConvertingWorker.run(OutputConvertingWorker.java:44) [io.airbyte-airbyte-workers-0.12.1-alpha.jar:?]
at io.airbyte.workers.wrappers.JobOutputSyncWorker.run(JobOutputSyncWorker.java:32) [io.airbyte-airbyte-workers-0.12.1-alpha.jar:?]
at io.airbyte.scheduler.WorkerRun.lambda$new$0(WorkerRun.java:53) [io.airbyte-airbyte-scheduler-0.12.1-alpha.jar:?]
at io.airbyte.scheduler.WorkerRun.call(WorkerRun.java:61) [io.airbyte-airbyte-scheduler-0.12.1-alpha.jar:?]
at io.airbyte.scheduler.WorkerRun.call(WorkerRun.java:42) [io.airbyte-airbyte-scheduler-0.12.1-alpha.jar:?]
at io.airbyte.commons.concurrency.LifecycledCallable.execute(LifecycledCallable.java:114) [io.airbyte-airbyte-commons-0.12.1-alpha.jar:?]
at io.airbyte.commons.concurrency.LifecycledCallable.call(LifecycledCallable.java:98) [io.airbyte-airbyte-commons-0.12.1-alpha.jar:?]
at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
at java.lang.Thread.run(Thread.java:832) [?:?]
Suppressed: io.airbyte.workers.WorkerException: Tap process wasn't successful
at io.airbyte.workers.normalization.DefaultNormalizationRunner.close(DefaultNormalizationRunner.java:100) ~[io.airbyte-airbyte-workers-0.12.1-alpha.jar:?]
at io.airbyte.workers.DefaultSyncWorker.run(DefaultSyncWorker.java:109) [io.airbyte-airbyte-workers-0.12.1-alpha.jar:?]
at io.airbyte.workers.DefaultSyncWorker.run(DefaultSyncWorker.java:48) [io.airbyte-airbyte-workers-0.12.1-alpha.jar:?]
at io.airbyte.workers.wrappers.OutputConvertingWorker.run(OutputConvertingWorker.java:44) [io.airbyte-airbyte-workers-0.12.1-alpha.jar:?]
at io.airbyte.workers.wrappers.JobOutputSyncWorker.run(JobOutputSyncWorker.java:32) [io.airbyte-airbyte-workers-0.12.1-alpha.jar:?]
at io.airbyte.scheduler.WorkerRun.lambda$new$0(WorkerRun.java:53) [io.airbyte-airbyte-scheduler-0.12.1-alpha.jar:?]
at io.airbyte.scheduler.WorkerRun.call(WorkerRun.java:61) [io.airbyte-airbyte-scheduler-0.12.1-alpha.jar:?]
at io.airbyte.scheduler.WorkerRun.call(WorkerRun.java:42) [io.airbyte-airbyte-scheduler-0.12.1-alpha.jar:?]
at io.airbyte.commons.concurrency.LifecycledCallable.execute(LifecycledCallable.java:114) [io.airbyte-airbyte-commons-0.12.1-alpha.jar:?]
at io.airbyte.commons.concurrency.LifecycledCallable.call(LifecycledCallable.java:98) [io.airbyte-airbyte-commons-0.12.1-alpha.jar:?]
at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
at java.lang.Thread.run(Thread.java:832) [?:?]
```

This is the generated DBT model:
```
root@ca0be5786d74:/tmp/workspace/26/0/normalize/models/generated# cat email_events.sql
with
email_events_node as (
  select
    _airbyte_emitted_at,
    {{ dbt_utils.current_timestamp_in_utc()  }} as _airbyte_normalized_at,
    cast({{ json_extract_scalar('_airbyte_data', ['id'])  }} as {{ dbt_utils.type_string()  }}) as id,
    cast({{ json_extract_scalar('_airbyte_data', ['url'])  }} as {{ dbt_utils.type_string()  }}) as url,
    cast({{ json_extract_scalar('_airbyte_data', ['from'])  }} as {{ dbt_utils.type_string()  }}) as from,
    cast({{ json_extract_scalar('_airbyte_data', ['hmid'])  }} as {{ dbt_utils.type_string()  }}) as hmid,
    cast({{ json_extract_scalar('_airbyte_data', ['type'])  }} as {{ dbt_utils.type_string()  }}) as type,
    cast({{ json_extract_scalar('_airbyte_data', ['appId'])  }} as {{ dbt_utils.type_float()  }}) as appId,
    cast({{ json_extract_scalar('_airbyte_data', ['linkId'])  }} as {{ dbt_utils.type_float()  }}) as linkId,
    cast({{ json_extract_scalar('_airbyte_data', ['smtpId'])  }} as {{ dbt_utils.type_string()  }}) as smtpId,
    cast({{ json_extract_scalar('_airbyte_data', ['appName'])  }} as {{ dbt_utils.type_string()  }}) as appName,
    cast({{ json_extract_scalar('_airbyte_data', ['created'])  }} as {{ dbt_utils.type_string()  }}) as created,
    cast({{ json_extract_scalar('_airbyte_data', ['subject'])  }} as {{ dbt_utils.type_string()  }}) as subject,
    cast({{ json_extract_scalar('_airbyte_data', ['duration'])  }} as {{ dbt_utils.type_float()  }}) as duration,
    cast({{ json_extract_scalar('_airbyte_data', ['portalId'])  }} as {{ dbt_utils.type_float()  }}) as portalId,
    cast({{ json_extract_scalar('_airbyte_data', ['response'])  }} as {{ dbt_utils.type_string()  }}) as response,
    cast({{ json_extract_scalar('_airbyte_data', ['ipAddress'])  }} as {{ dbt_utils.type_string()  }}) as ipAddress,
    cast({{ json_extract_scalar('_airbyte_data', ['recipient'])  }} as {{ dbt_utils.type_string()  }}) as recipient,
    cast({{ json_extract_scalar('_airbyte_data', ['userAgent'])  }} as {{ dbt_utils.type_string()  }}) as userAgent,
    cast({{ json_extract_scalar('_airbyte_data', ['deviceType'])  }} as {{ dbt_utils.type_string()  }}) as deviceType,
    {{ cast_to_boolean(json_extract_scalar('_airbyte_data', ['filteredEvent']))  }} as filteredEvent,
    cast({{ json_extract_scalar('_airbyte_data', ['emailCampaignId'])  }} as {{ dbt_utils.type_float()  }}) as emailCampaignId,
    cast({{ json_extract_scalar('_airbyte_data', ['emailCampaignGroupId'])  }} as {{ dbt_utils.type_float()  }}) as emailCampaignGroupId
  from {{ source('public', '_airbyte_raw_email_events')  }}
),
email_events_with_id as (
  select
    *,
    {{ dbt_utils.surrogate_key(['id',
        'url',
        'from',
        'hmid',
        'type',
        'appId',
        'linkId',
        'smtpId',
        'appName',
        'created',
        'subject',
        'duration',
        'portalId',
        'response',
        'ipAddress',
        'recipient',
        'userAgent',
        'deviceType',
        boolean_to_varchar('filteredEvent'),
        'emailCampaignId',
        'emailCampaignGroupId'])  }} as _airbyte_email_events_hashid
    from email_events_node
)
select * from email_events_with_id
```

And the final:
```
root@ca0be5786d74:/tmp/workspace/26/0/build/run/airbyte_utils/models/generated# cat email_events.sql


  create  table "postgres"."public"."email_events__dbt_tmp"
  as (
    with
email_events_node as (
  select
    _airbyte_emitted_at,

    (current_timestamp at time zone 'utc')::
    timestamp

 as _airbyte_normalized_at,
    cast(jsonb_extract_path_text("_airbyte_data",'id') as
    varchar
) as id,
    cast(jsonb_extract_path_text("_airbyte_data",'url') as
    varchar
) as url,
    cast(jsonb_extract_path_text("_airbyte_data",'from') as
    varchar
) as from,
    cast(jsonb_extract_path_text("_airbyte_data",'hmid') as
    varchar
) as hmid,
    cast(jsonb_extract_path_text("_airbyte_data",'type') as
    varchar
) as type,
    cast(jsonb_extract_path_text("_airbyte_data",'appId') as
    float
) as appId,
    cast(jsonb_extract_path_text("_airbyte_data",'linkId') as
    float
) as linkId,
    cast(jsonb_extract_path_text("_airbyte_data",'smtpId') as
    varchar
) as smtpId,
    cast(jsonb_extract_path_text("_airbyte_data",'appName') as
    varchar
) as appName,
    cast(jsonb_extract_path_text("_airbyte_data",'created') as
    varchar
) as created,
    cast(jsonb_extract_path_text("_airbyte_data",'subject') as
    varchar
) as subject,
    cast(jsonb_extract_path_text("_airbyte_data",'duration') as
    float
) as duration,
    cast(jsonb_extract_path_text("_airbyte_data",'portalId') as
    float
) as portalId,
    cast(jsonb_extract_path_text("_airbyte_data",'response') as
    varchar
) as response,
    cast(jsonb_extract_path_text("_airbyte_data",'ipAddress') as
    varchar
) as ipAddress,
    cast(jsonb_extract_path_text("_airbyte_data",'recipient') as
    varchar
) as recipient,
    cast(jsonb_extract_path_text("_airbyte_data",'userAgent') as
    varchar
) as userAgent,
    cast(jsonb_extract_path_text("_airbyte_data",'deviceType') as
    varchar
) as deviceType,
    cast(jsonb_extract_path_text("_airbyte_data",'filteredEvent') as boolean) as filteredEvent,
    cast(jsonb_extract_path_text("_airbyte_data",'emailCampaignId') as
    float
) as emailCampaignId,
    cast(jsonb_extract_path_text("_airbyte_data",'emailCampaignGroupId') as
    float
) as emailCampaignGroupId
  from "postgres".public._airbyte_raw_email_events
),
email_events_with_id as (
  select
    *,
    md5(cast(

    coalesce(cast(id as
    varchar
), '') || '-' || coalesce(cast(url as
    varchar
), '') || '-' || coalesce(cast(from as
    varchar
), '') || '-' || coalesce(cast(hmid as
    varchar
), '') || '-' || coalesce(cast(type as
    varchar
), '') || '-' || coalesce(cast(appId as
    varchar
), '') || '-' || coalesce(cast(linkId as
    varchar
), '') || '-' || coalesce(cast(smtpId as
    varchar
), '') || '-' || coalesce(cast(appName as
    varchar
), '') || '-' || coalesce(cast(created as
    varchar
), '') || '-' || coalesce(cast(subject as
    varchar
), '') || '-' || coalesce(cast(duration as
    varchar
), '') || '-' || coalesce(cast(portalId as
    varchar
), '') || '-' || coalesce(cast(response as
    varchar
), '') || '-' || coalesce(cast(ipAddress as
    varchar
), '') || '-' || coalesce(cast(recipient as
    varchar
), '') || '-' || coalesce(cast(userAgent as
    varchar
), '') || '-' || coalesce(cast(deviceType as
    varchar
), '') || '-' || coalesce(cast(filteredEvent as
    varchar
), '') || '-' || coalesce(cast(emailCampaignId as
    varchar
), '') || '-' || coalesce(cast(emailCampaignGroupId as
    varchar
), '')

 as
    varchar
)) as _airbyte_email_events_hashid
    from email_events_node
)
select * from email_events_with_id
  );
```

## Severity of the bug for you
High

## Airbyte Version
at least 0.8.0 through 0.12.1

**Comment by jrhizor** [comment]
It's possible that this just is fully addressed by https://github.com/airbytehq/airbyte/pull/1753 even though the exception was different than what was listed in https://github.com/airbytehq/airbyte/issues/1561. Going to check.

**Comment by jrhizor** [comment]
Looks like this is actually solved by that. I can get it to sync correctly on the latest master (not the latest published version).

**Comment by sherifnada** [comment]
@jrhizor this can be fixed with a patch release

**Comment by jrhizor** [comment]
Yup, going to do that as soon as the build fix Charles just pushed passes.

**Comment by jrhizor** [comment]
Fixed in v0.13.0
